.jenkins/Jenkinsfile

```diff
@@ -22,517 +22,447 @@
+    // retry(3) // Do we need it?
+  }
+  parameters {
+    string(name: 'repository', defaultValue: 'https://github.com/HadesArchitect/cassandra', description: 'Cassandra Repository')
+    string(name: 'branch', defaultValue: '*/aleks-cassius', description: 'Branch')
+    choice(name: 'architecture', choices: archsSupported + "all", description: 'Pick architecture. The ARM64 is disabled by default at the moment.')
+    choice(name: 'jdk', choices: jdksSupported + "all", description: 'Pick JDK versions.')
+    booleanParam(name: 'stage_artifacts', defaultValue: false, description: 'Disable to exlude stage') // return default to true
+    booleanParam(name: 'stage_lint', defaultValue: false) // return default to true
+    booleanParam(name: 'stage_debian', defaultValue: false) // return default to true
+    booleanParam(name: 'stage_redhat', defaultValue: false) // return default to true
+    booleanParam(name: 'stage_fqltool-test', defaultValue: false) // return default to true
+    booleanParam(name: 'stage_cqlsh-test', defaultValue: false) // return default to true
+    booleanParam(name: 'stage_test-cdc', defaultValue: false) // return default to true
+    booleanParam(name: 'stage_test', defaultValue: false) // return default to true
+    booleanParam(name: 'stage_test-trie', defaultValue: false) // return default to true
+    booleanParam(name: 'stage_test-compression', defaultValue: false) // return default to true
+    booleanParam(name: 'stage_stress-test', defaultValue: false) // return default to true
+    booleanParam(name: 'stage_test-burn', defaultValue: false) // return default to true
+    booleanParam(name: 'stage_long-test', defaultValue: false) // return default to true
+    booleanParam(name: 'stage_test-oa', defaultValue: false) // return default to true
+    booleanParam(name: 'stage_test-system-keyspace-directory', defaultValue: false) // return default to true
+    booleanParam(name: 'stage_jvm-dtest', defaultValue: false) // return default to true
+    booleanParam(name: 'stage_jvm-dtest-upgrade', defaultValue: false) // return default to true
+    booleanParam(name: 'stage_dtest', defaultValue: false) // return default to true
+    booleanParam(name: 'stage_dtest-novnode', defaultValue: false) // return default to true
+    booleanParam(name: 'stage_dtest-offheap', defaultValue: false) // return default to true
+    booleanParam(name: 'stage_dtest-large', defaultValue: false) // return default to true
+    booleanParam(name: 'stage_dtest-large-novnode', defaultValue: false) // return default to true
+    booleanParam(name: 'stage_dtest-upgrade', defaultValue: false) // return default to true
+    booleanParam(name: 'stage_dtest-upgrade-novnode', defaultValue: false) // return default to true
+    booleanParam(name: 'stage_dtest-upgrade-large', defaultValue: false) // return default to true
+    booleanParam(name: 'stage_dtest-upgrade-novnode-large', defaultValue: false) // return default to true
```

Comment:
given the stages (and test types) are (/will be) dynamic, this booleanParam approach won't survive.

can we use either a `choice` or `stringParam` (the string param being like a regex) ?

Reply:
I planned to generate booleanParams based on the `stepsMap` but it's hard with Jenkins, need an additional plugin or some dark magic. Let's stick to stringParam for now, I guess.